halo1/model.py

```diff
@@ -108,13 +213,30 @@ def import_halo1_markers_from_jms(jms, *, scale=1.0, node_size=0.01,
 		if import_radius:
 			set_uniform_scale(scene_marker, marker.radius)
 
-		# Assign parent if index is valid.
-		scene_marker.parent = scene_nodes.get(marker.parent, None)
-
 		set_rotation_from_jms(scene_marker, marker)
 
 		set_translation_from_jms(scene_marker, marker, scale)
 
+		# Assign parent if index is valid.
+		parent = scene_nodes.get(marker.parent, None)
+		if armature and parent:
+			scene_marker.parent = armature
+			scene_marker.parent_type = 'BONE'
+			scene_marker.parent_bone = parent.name
+
+			# The rotation is offset by 0, 0, -90 euler, so we rotate it to
+			# fix that.
+
+			scene_marker.location.rotate(Euler((0.0, 0.0, math.radians(90.0))))
+
+			# Then we subtract the length of the parent bone from the y axis,
+			# the y axis being the axis that is offset.
+
+			# Halo bone behavior expects markers their 0 0 0 to be at the bone
+			# base. But when attaching in Blender its 0 0 0 is instead at the
+			# bone's tail end.
```

Comment:
This wording is kinda grammatically funky and I'm not exactly sure what is being said here.